tiledb/common/macros.h

```diff
@@ -56,4 +56,31 @@
   DISABLE_MOVE(C);                      \
   DISABLE_MOVE_ASSIGN(C)
 
+/** Disables warning for specific compiler
+ */
+#if defined(_MSC_VER)
```

Comment:
You've got definitions for MSVC but not using them. It would be better either to do more (implement for MSVC) or less (don't implement for MSVC at all), but not this in the middle.